src/query/service/src/interpreters/access/privilege_access.rs

```diff
@@ -520,6 +523,34 @@ impl PrivilegeAccess {
         Ok(())
     }
 
+    async fn validate_table_function_access(&self, table_func_name: &str) -> Result<()> {
+        if SYSTEM_TABLE_FUNCTIONS.iter().any(|x| x == &table_func_name) {
+            // need Super privilege to invoke system table functions
+            let privilege = UserPrivilegeType::Super;
+            let session = self.ctx.get_current_session();
+            let current_user = self.ctx.get_current_user()?;
+            session
```

Comment:
just `session.validate_privilege(&GrantObject::Global, privilege, true).await` ?

i suppose that the error message has already been formated as the blow inside this `validate_privilege` function

Reply:
The "default" error message is `ERROR 1105 (HY000): PermissionDenied. Code: 1063, Text = Permission denied.`, which is not very informative, IMO.